### DISCUSSION TITLE
Arotherm Plus Heat Pump - variables for working out COP

### DISCUSSION THREAD

**Discussion by zarch1972** [john30/ebusd#903] [Q&A]
On my 5Kw Arotherm Plus I have a Sontex heat meter so that I can track the heat pump and work out instant / historical COP.

https://emoncms.org/energystatsuk

I was wondering whether I could pull the desired variables out of the ebusd config for real-time monitoring?

- Flow Temperature - (sensor.ebusd_hmu_status01_temp1?)
- Return Temperature - (sensor.ebusd_hmu_status01_temp1_2?)
- Input Electrical Power - (ebusd hmu CurrentConsumedPower?)
- Flow rate - (can't find this, is it there?)

If I have these values, I can work out COP.

Heat Output = flow rate x SHC (4.2 for water) x DT (Flow - Return)

Then coupled with Electric Input I can do COP..... (Heat Output / Electric Input)

And then i'd like to compare the results from ebusd versus the heat meter

Do I have to provide any special flag / config file in ebusd to get all the Arotherm variables out?

This is what I'm seeing now

```
ebusdctl find
scan.08  = Vaillant;HMU00;0607;5103
scan.08 id = 21;22;34;0010021131;1610;006001;N4
scan.15  = Vaillant;BASV0;0217;8803
scan.15 id = 21;20;47;0020262148;0082;015641;N9
scan.76  = Vaillant;VWZIO;0202;0103
scan.76 id = 21;22;37;0010031644;0082;009299;N5
scan.ec  = Vaillant;SOL00;0217;8803
scan.ec id = 21;20;47;0020262148;0082;015641;N9
scan.f6  = Vaillant;NETX2;4029;5703

```
```
ebusdctl info
update: 10
address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=HMU00;SW=0607;HW=5103", loaded "vaillant/08.hmu.csv"
address 10: master #2
address 15: slave #2, scanned "MF=Vaillant;ID=BASV0;SW=0217;HW=8803", loaded "vaillant/15.basv.csv"
address 31: master #8, ebusd
address 36: slave #8, ebusd, scanning
address 71: master #9
address 76: slave #9, scanned "MF=Vaillant;ID=VWZIO;SW=0202;HW=0103"
address ec: slave, scanned "MF=Vaillant;ID=SOL00;SW=0217;HW=8803", loaded "vaillant/ec.sol.sc.csv"
address f1: master #10
address f6: slave #10, scanned "MF=Vaillant;ID=NETX2;SW=4029;HW=5703"
```

Are either of these flow rate?
```
hmu WaterThroughput = no data stored
sc FlowRate = no data stored
```
if so, how do I get them recording data?

Any help would be appreciated.

**Comment by cyberthom42** [comment]
I think you are using an old csv. Look at the discussions, many new values were added. I think they have not been updated to the main csv so far. Which values are you looking for?

Water Throughput:
r,,HE_Durchfluss,T.0.43,,,,052b03ffff,,,UIN,,l/h,Durchfluss,,,,,,,,,,,,,,,,,,,,,,,,,,,

You can also have a look at my github, we actually compared some values here and made some calculation (German). 
https://github.com/cyberthom42/vaillant-arotherm-plus/discussions/2

There is also a growing wiki at which we collect Arotherm information:
https://github.com/cyberthom42/vaillant-arotherm-plus/wiki

You are welcome to contribute

**Reply by zarch1972** [reply]
Thank you for the reply.

I just installed the latest 23.1 from the package, so will be using whatever CSV came with that. 
When do additions get added to the master CSV @john30 ?

Where are the latest csv stored?

Values I'm after

- Flow Temp
- Return Temp
- Flow Rate
- Electric Input

As you have done, I can work COP and various things out with those raw values.

Thanks.

**Comment by cyberthom42** [comment]
FlowTemp: there are different ways to get this value, some of them are mor accurate than others. I am using the following:
r5,hmu,FlowTemp,,,08,b51a,05ff3220,,s,IGN:3,,,,,s,D2C,16,,
r1,hmu,HE_Vorlauftemperatur,T.0.40,,08,b514,052803ffff,,s,IGN:2,,,,,s,UIN,10,tempv,Vorlauftemperatur
r5,vwzio,HE_Vorlauftemperatur_Heizstab,T.1.130,,76,b514,058203ffff,,s,IGN:2,,,,,s,UIN,10,tempv,Vorlauftemperatur Heizstab
r1,basv,HwcFlowTemp,HwcFlowTemp,,15,b524,020001000800,,s,IGN:4,,,,tempv,s,EXP,,°C,desired flow temp for Hwc

Return Temp:
r1,hmu,HE_Ruecklauftemperatur,T.0.41,,08,b514,052903ffff,,s,IGN:2,,,,,s,UIN,10,tempv,Ruecklauftemperatur

Flow Rate:
 r,,HE_Durchfluss,T.0.43,,,,052b03ffff,,,UIN,,l/h,Durchfluss,,,,,,,,,,,,,,,,,,,,,,,,,,,

Electric Input:
r,hmu,CurrentConsumedPower,,,08,b51a,05ff3224,,s,IGN:3,,,,,s,D1B,10,kW,aktueller Stromverbrauch der Wärmepumpe in kW
=> The value from the Arotherm is not very accuryte, especially at low compressor rates as it only increases in steps of 100W

**Comment by zarch1972** [comment]
Just coming back to this after playing for a while.

It looks like i've found usable flow/return and indoor/outdoor temps from my 5kW Arotherm Plus

![image](https://github.com/john30/ebusd/assets/23335221/b931e9e6-de56-4171-8eb6-d3672cd900cb)

But all of Flow rate (hmu_buildingcircuitflow), currentconsumedpower and currentyieldpower don't update often enough to be usable?

You can see on the graphs the realtime values i'm pulling from my electric and heat meters alongside the ebusd values.
I'm trying to see how well the ebusd values match the electric/heat meter.

Is there anyway of increasing the refresh times of flow rate, currentconsumedpower and currentyieldpower?

Or different fields to look at?

**Reply by CommanderROR9** [reply]
I am just getting started with this. I used a .csv from a forked Repo to get more relevant data, but I don't think I have the building flow rate...any chance you can tell me which .csv you are using?

**Reply by zarch1972** [reply]
I've been playing with

https://github.com/jonesPD/ebusd-configuration

**Reply by CommanderROR9** [reply]
Thanks. Will give it a try.

**Reply by CommanderROR9** [reply]
Hmm...I gave it a go, but it doesn't seem ro work very well with my system. Maybe I have to do a mpre thorough rescan
